src/appleseed/renderer/kernel/rendering/generic/genericsamplerenderer.cpp

```diff
@@ -218,34 +219,37 @@ namespace
                 if (iterations == 1)
                 {
                     // Shade the first intersection point along the ray.
-                    m_shading_engine.shade(
+                    // Stop if shade returns false.
```

Comment:
The comment on line 250 ("Stop if we found a matte object") is more informative, even though I don't know if it's quite accurate or future-proof enough (maybe `shade()` will return `false` in other cases in the future?).

In any case it should be the same comment, and the comment should apply to the `if` below.